Expanded capabilities

whobday opened the pull request:
- enumerates and emits Choices/Optionsets/Picklists
 - enumerates and emits fields and datatypes
 - enumerates and emits relationships
 - switched relationships notation to inline

Commit:
Expanded capabilities
 - enumerates and emits Choices/Optionsets/Picklists
 - enumerates and emits fields and datatypes
 - enumerates and emits relationships
 - switched relationships notation to inline

YonatanCohavi commented on ERDMaker/Services/CRMService.cs:
```diff
@@ -22,59 +32,141 @@ public static ICollection<string> GetAllEntities(IOrganizationService service)
+
+        private static void DefinePrimaryKey(EntityMetadata metadata, Table table)
+        {
+            var field = metadata.Attributes.Single(f => f.IsPrimaryId == true);
```
this line can return multiple attributes. try the contact entity.

YonatanCohavi commented on ERDMaker/Services/DiagramBuilder.cs:
```diff
@@ -2,37 +2,57 @@
-                {
-                    references.Add($"Ref: {table.Name}.{field.Name} > {reference}");
-                }
+                optionSetBuilder.AppendLine($"\t\"{value}\"");
```
there is a bug when the optionset label contains quotation mark

whobday replied:
There is no way to escape the quotation mark in the DBML spec.  We'll have to settle for a simple replacement to a single quote

YonatanCohavi commented:
hey thanks for the pull request! 😀.
I have some issues with these changes. this makes the ERD very crowded. the main idea was to understand the relationships between the selected tables. 
Please add these as optional fields in the main window.

whobday commented:
I'll make the field generation optional

Commit:
Suggested Updates
 - Added options to generae fields, optionsets, and relationships
 - Updated PrimaryKey lookup
 - Replace " with ' for enum values

whobday commented:
The updates are ready to go
